test/haxesharp/collections/DictionaryTest.hx

```diff
@@ -0,0 +1,253 @@
+package haxesharp.collections;
+
+using haxesharp.collections.Dictionary;
+using haxesharp.test.Assert;
+using haxesharp.exceptions.ArgumentException;
+using haxesharp.exceptions.ArgumentNullException;
+using haxesharp.exceptions.KeyNotFoundException;
+
+class DictionaryTest
+{
+    @Test
+    public function dictionaryAllowsIntKeys()
+    {
+        testDictionaryGetSetAndRemove(1, 1, 7);
+        testDictionaryMapConstructor([
```

Comment:
I think it's really interesting how you refactored out all the common code in a type-safe way (I couldn't figure this out). The test is not easy to read as a result, but I think I'm okay with that trade-off.